### PR TITLE
Dockerize application

### PR DESCRIPTION
**Pull request by NICHTJ3**
### Solves:
* #52
* #2
* #103 

__Notes:__
This solves most of the issues with getting up and running it also adds most of the setup required for docker swarm but volumes will not scale when deployed in a swarm and when requesting data through nginx to the API you will need to ensure you have a trailing slash I was unable to figure out a way around this.

* There will need to be a database service added to this when we start using one
* Run.py is now gone

### CODE REVIEW

**Comment by NICHTJ3** [comment]
Instead of having env variables in the compose files i could have probably pointed to the .env files. What do you think?

**Comment by NICHTJ3** [comment]
I have added the db service but i might change the compose file so that all services talk on a docker network and the only public port will be the nginx one

**Comment by NICHTJ3** [comment]
Also we need to think of a way to hide the database passwords in the compose files

File: back-end/Dockerfile-prod [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# base image
+FROM kennethreitz/pipenv
```
Comment (Zimbardoex) [inline back-end/Dockerfile-prod]: Should we be using someone else's image here? Does the one you created above work?

**Comment by NICHTJ3** [comment]
We should probably solve #103 in here aswell

**Pull request merged**